src/vs/workbench/browser/quickopen.ts

```diff
@@ -247,20 +248,30 @@ export class EditorQuickOpenEntry extends QuickOpenEntry implements IEditorQuick
 	}
 
 	public run(mode: Mode, context: IEntryRunContext): boolean {
-		if (mode === Mode.OPEN) {
+		const hideWidget = mode === Mode.OPEN;
+
+		let opts = this.getOptions();
+		if (mode === Mode.OPEN_BEHIND) {
+			opts = objects.assign(opts || {}, {
+				pinned: true,
+				revealIfVisible: true,
+				preserveFocus: true
+			});
+		}
+
+		if (mode === Mode.OPEN || mode === Mode.OPEN_BEHIND) {
 			let sideBySide = context.keymods.indexOf(KeyMod.CtrlCmd) >= 0;
 
 			let input = this.getInput();
 			if (input instanceof EditorInput) {
-				this.editorService.openEditor(input, this.getOptions(), sideBySide).done(null, errors.onUnexpectedError);
+				this.editorService.openEditor(input, opts, sideBySide).done(null, errors.onUnexpectedError);
 			} else {
+				objects.assign((<IResourceInput>input).options, opts);
```

Comment:
@wprater also you seem to alter the way options used to be handled with `IResourceInput`. you are merging the options from getOptions() call into IResourceInput. we did not do that before though:

`this.editorService.openEditor(<IResourceInput>input, sideBySide).done(null, errors.onUnexpectedError);`